replication/row_event.go

```diff
@@ -257,13 +259,15 @@ func (e *RowsEvent) Decode(data []byte) error {
 	var err error
 
 	// ... repeat rows until event-end
-	// Why using pos < len(data) - 1 instead of origin pos < len(data) to check?
-	// See mysql
-	//  https://github.com/mysql/mysql-server/blob/5.7/sql/log_event.cc#L9006
-	//	https://github.com/mysql/mysql-server/blob/5.7/sql/log_event.cc#L2492
-	// A user panics here but he can't give me more information, and using len(data) - 1
-	// fixes this panic, so I will try to construct a test case for this later.
-	for pos < len(data)-1 {
+	defer func() {
+		// Some people meet panic when we use `pos < len(data)`, but using `pos < len(data) - 1`
+		// before also misses data, see Decode TestLastNull suite.
+		// So we log detailed information to debug here and terminate the program.
+		if r := recover(); r != nil {
+			log.Fatalf("parse rows event panic %v, data %q, parsed rows %#v, table map %#v", r, data, e, e.Table)
+		}
+	}()
+	for pos < len(data) {
```

Comment:
Can this fix the panic problem? I am confused that seems this PR only add the trace log.